src/ResultsPage.js

```diff
@@ -117,23 +128,49 @@ class ResultsPage extends Component {
         let breweryIbus = brewery.beers.map(beer => {
           return `${Math.round(beer.ibu / 10) * 10}-${(Math.round(beer.ibu / 10) * 10) + 9}`;
         });
-        if(breweryIbus.includes(this.state.ibuSelection)) {
+        if (breweryIbus.includes(this.state.ibuSelection)) {
           acc.push(brewery);
         }
         return acc;
-      }, [])
-      this.setState({filteredBreweries: filterResults}); //add filter by ABV to callback
+      }, []);
+      this.setState({ filteredBreweries: filterResults }, () => this.filterByStarred());
+    } else {
+      this.filterByStarred();
+    }
+  }
+
+  updateStarredList = (name, change) => {
+    // let list = JSON.parse(localStorage.getItem('userStarredBreweries')) || [];
+    let list = this.state.starredBreweries ? [...this.state.starredBreweries] : [];
```

Comment:
We already know that `this.state.starredBreweries` exists because you have made it so with the conditional/fallback you're doing on line 7. Just set your `list` variable to `this.state.starredBreweries` -- it will either represent an array of elements from localStorage, or an empty array.

Reply:
Sounds good.  We had it without the conditional before, but there's some extra clutter from my desperate attempts to be more specific haha.

QQ for you, if we assign the starredBreweries property to the variable without the spread operator, does that let us interact with it directly with push, splice, etc so we can save an updated version to local storage?  Still a little fuzzy on copy vs deep copy vs link to on arrays and have had a hard time finding the right info online.

Reply:
Yep you should still be able to interact with the array and all its prototype methods without having to spread. Because you're doing a `JSON.parse()` on line 7 for your localStorage data, it's automatically turning into that array for you. If you *didn't* do the JSON.parse(), that's where you would run into issues trying to use array prototype methods on the value, because it would be returning a string for you instead.

Reply:
Amazing, that did it.  Thank you SO much!